lib/kvdb/kvdb_meta.c

```diff
@@ -573,8 +573,9 @@ kvdb_meta_deserialize(struct kvdb_meta *const meta, const char *const kvdb_home)
         goto out;
     }
     version_val = cJSON_GetNumberValue(version);
-    if (round(version_val) != version_val) {
-        log_err("'version' key in %s/kvdb.meta is not a whole number", kvdb_home);
+    if (round(version_val) != version_val || version_val < 0 || version_val > UINT_MAX) {
+        log_err("'version' key in %s/kvdb.meta must be a whole number between 0 and %u",
+            kvdb_home, UINT_MAX);
```

Comment:
The existance of this code tells you what a botch cJSON's "Number" type is...
I would think if we ever run into an ILP64 model machine then UINT_MAX would be 64 bits?  So maybe you want to use UINT32_MAX here if that's what's your're expecting...

Reply:
I am using uint because I know that the enum being represented here will fit well within that integer type.

Reply:
Point is you know that now, but someone modifying the enum won't know to come look at and fix this code...

Reply:
static assert incoming